src/Mod/PartDesign/Gui/TaskPocketParameters.cpp

```diff
@@ -291,6 +353,193 @@ void TaskPocketParameters::onOffsetChanged(double len)
     recomputeFeature();
 }
 
+void TaskPocketParameters::fillDirectionCombo()
+{
+    bool oldVal_blockUpdate = blockUpdate;
+    blockUpdate = true;
+
+    if (axesInList.empty()) {
+        ui->directionCB->clear();
+        // add sketch normal
+        PartDesign::ProfileBased* pcFeat = static_cast<PartDesign::ProfileBased*>(vp->getObject());
+        Part::Part2DObject* pcSketch = dynamic_cast<Part::Part2DObject*>(pcFeat->Profile.getValue());
+        if (pcSketch)
+            addAxisToCombo(pcSketch, "N_Axis", QObject::tr("Sketch normal"));
+        // add the other entries
+        addAxisToCombo(0, std::string(), tr("Select reference..."));
+        // we start with the sketch normal as proposal for the custom direction
+        if (pcSketch)
+            addAxisToCombo(pcSketch, "N_Axis", QObject::tr("Custom direction"));
+    }
+
+    // add current link, if not in list
+    // first, figure out the item number for current axis
+    int indexOfCurrent = -1;
+    App::DocumentObject* ax = propReferenceAxis->getValue();
+    const std::vector<std::string>& subList = propReferenceAxis->getSubValues();
+    for (size_t i = 0; i < axesInList.size(); i++) {
+        if (ax == axesInList[i]->getValue() && subList == axesInList[i]->getSubValues()) {
+            indexOfCurrent = i;
+            break;
+        }
+    }
+    // if the axis is not yet listed in the combobox
+    if (indexOfCurrent == -1 && ax) {
+        assert(subList.size() <= 1);
+        std::string sub;
+        if (!subList.empty())
+            sub = subList[0];
+        addAxisToCombo(ax, sub, getRefStr(ax, subList));
+        indexOfCurrent = axesInList.size() - 1;
+        // the axis is not the normal, thus enable along direction
+        ui->checkBoxAlongDirection->setEnabled(true);
+        // we don't have custom direction thus disable its settings
+        ui->XDirectionEdit->setEnabled(false);
+        ui->YDirectionEdit->setEnabled(false);
+        ui->ZDirectionEdit->setEnabled(false);
+    }
+
+    // highlight either current index or set custom direction
+    PartDesign::Pocket* pcPocket = static_cast<PartDesign::Pocket*>(vp->getObject());
+    bool hasCustom = pcPocket->UseCustomVector.getValue();
+    if (indexOfCurrent != -1 && !hasCustom)
+        ui->directionCB->setCurrentIndex(indexOfCurrent);
+    if (hasCustom)
+        ui->directionCB->setCurrentIndex(2);
+
+    blockUpdate = oldVal_blockUpdate;
+}
+
+void TaskPocketParameters::addAxisToCombo(App::DocumentObject* linkObj,
+    std::string linkSubname, QString itemText)
+{
+    this->ui->directionCB->addItem(itemText);
+    this->axesInList.emplace_back(new App::PropertyLinkSub);
+    App::PropertyLinkSub& lnk = *(axesInList.back());
+    lnk.setValue(linkObj, std::vector<std::string>(1, linkSubname));
+}
+
+void TaskPocketParameters::onDirectionCBChanged(int num)
+{
+    PartDesign::Pocket* pcPocket = static_cast<PartDesign::Pocket*>(vp->getObject());
+
+    if (axesInList.empty() || !pcPocket)

```

Comment:
Is it actually possible for `pcPocket` to be null here? You have this same basic structure in several other callbacks in which you don't check for null.

Reply:
> Is it actually possible for pcPocket to be null here?

No. So this can go for consistency reasons. Done now.